### DISCUSSION TITLE
Rating: half star support (proposal)

### DISCUSSION THREAD

**Discussion by jmckerral** [primefaces/community#1245] [PrimeVue]
Hey!

While using the Rating component, I observed that the behaviour for non-integers is to simply truncate any decimal points.  For example, ratings of 4.2, 4.5, and 4.9 will all be shown as four stars.

I propose we add support to display half stars.  This can be activated with an optional boolean property that defaults to false so that existing behaviour is preserved (similar to `readonly`).

We should also add `halficon` (or similar) support into the template/slots.

We may also wish to add some sort of rounding behaviour, which would either apply when activating half stars, or could be another new property.

Excluded:
* Half value input.  Propose the change is for display only.  Perhaps the option could be active only in `readonly` mode?  Otherwise the behaviour may be a little weird (see a half star, choose a different value, be unable to get the original value back).

I can make these changes - I just wanted to check for existing thoughts/previous attempts/suggestions/best practices/advice/etc

All thoughts/feelings/hopes/dreams are welcome.

Cheers,

James

**Comment by joaomarcelofm** [comment]
There was an attempt to do it a while back, but it was rejected and never made it to the roadmap, it looks like. 😞 
https://github.com/primefaces/primevue/pull/1144

**Reply by slavco86** [reply]
Yeah, 3 years after - still no feature, but the PR was closed with "We prefer to do it ourselves" comment :face_palm

**Reply by jmckerral** [reply]
oh, interesting, I didn't see that - thanks!  I would still gladly spend some time on this but, of course, I'd like the odds of that time not being wasted to be a little higher than I judge them currently to be (hence the speculative question 😅)

**Reply by slavco86** [reply]
Yeah, it's a shame this is not current functionality. I think the easiest solution would be just to get rid of the rounding logic inside the component and just keep the decimal part of the `value` when exposing it via scoped slots.

So, for example if I pass 3.5 value via the `v-model` inside the `#onicon` slot `value` prop - I'd get 1.5, 2.5 and 3.5, instead of as currently we get 1,2 and 3 🤔

**Reply by slavco86** [reply]
@cagataycivici , is it possible to look at this request again, please?
Seems like a fairly easy win and will cover a pretty standard and widespread use case for such a UI pattern, like rating. 🙏

**Reply by joaomarcelofm** [reply]
@tugcekucukoglu, do you know if this feature will ever be available or, perhaps more importantly, if the Prime team is open to a new PR to address this request?

**Reply by eybarta** [reply]
Has anyone been able to customize the Rating component somehow to allow half stars?
This is a must-have.

**Comment by jmckerral** [comment]
Heyo!  I finally got around to attempting this myself.  [A PR is raised here](https://github.com/primefaces/primevue/pull/6637).

As ever, all thoughts/questions/suggestions are welcome.
